command/snapshot/inspect/snapshot_inspect.go

```diff
@@ -56,27 +65,181 @@ func (c *cmd) Run(args []string) int {
 	}
 	defer f.Close()
 
-	meta, err := snapshot.Verify(f)
+	readFile, meta, err := snapshot.Read(nil, f)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error reading snapshot: %s", err))
+	}
```

Comment:
I think we need to remove the `readFile` before exiting. We can use something similar to what we moved to Restore:

```go
defer func() {
	if err := readFile.Close(); err != nil {
		c.UI.Error(fmt.Sprintf("Failed to close temp snapshot: %v", err))
	}
	if err := os.Remove(snap.Name()); err != nil {
		c.UI.Error(fmt.Sprintf("Failed to clean up temp snapshot: %v", err))
	}
}()
```